src/cudadev/CUDACore/host_unique_ptr.h

```diff
@@ -39,17 +39,17 @@ namespace cms {
     // Allocate pinned host memory
     template <typename T>
     typename host::impl::make_host_unique_selector<T>::non_array make_host_unique(cudaStream_t stream) {
-      static_assert(std::is_trivially_constructible<T>::value,
-                    "Allocating with non-trivial constructor on the pinned host memory is not supported");
+      static_assert(std::is_trivially_copyable<T>::value,
+                    "Allocating with non-trivial copy on the pinned host memory is not supported");
```

Comment:
This raises the question if `make_host_unique()` should value-initialize like `std::make_unique()`, Or if there should be a separate `make` function that does also the value initialization.

Reply:
In the current SoA examples, we are unaffected as we do a 2 stage allocation where `make_(host|device)_unique()` are used to provide a bare buffer which will typically be filled by a ES/ED producer. Actually, this is the most common use case in general.

We can work this around with placement new, but I'm not sure we have so many use cases. It also breaks some symmetry with the device memory (Although this is less of an issue as we tend to a model where CPU is just another device).